cmd/create/machinepool/cmd.go

```diff
@@ -207,6 +210,23 @@ func run(cmd *cobra.Command, _ []string) {
 		os.Exit(1)
 	}
 
+	// Validate flags that are only allowed for multi-AZ clusters
+	isMultiAvailabilityZoneSet := cmd.Flags().Changed("multi-availability-zone")
+	if isMultiAvailabilityZoneSet && !cluster.MultiAZ() {
+		reporter.Errorf("Setting the `multi-availability-zone` flag is only allowed for multi-AZ clusters")
+		os.Exit(1)
+	}
+	isAvailabilityZoneSet := cmd.Flags().Changed("availability-zone")
+	if isAvailabilityZoneSet && !cluster.MultiAZ() {
+		reporter.Errorf("Setting the `availability-zone` flag is only allowed for multi-AZ clusters")
+		os.Exit(1)
+	}
+
+	if isAvailabilityZoneSet && isMultiAvailabilityZoneSet && args.multiAvailabilityZone {
+		reporter.Errorf("Setting the `availability-zone` flag is only allowed for a single AZ machine pool")
```

Comment:
wdyt about _Setting the `availability-zone` flag is only supported for creating a single AZ machine pool in multi az cluster "_   we need cluster type reference